sklearn/covariance/_shrunk_covariance.py

```diff
@@ -325,31 +348,13 @@ def ledoit_wolf(X, *, assume_centered=False, block_size=1000):
 
```

Comment:
We discussed this with @jeremiedbb in another PR. He suggested actually doing the double validation and adding the function to the list to run the common test.

Sorry for the back and forth. We did not anticipate all the issues. We start to settle on the way we want the validation to be done :)

Reply:
When I added function to public tests it raised this error

> parameter_constraints = getattr(func, "_skl_parameter_constraints")
> E       AttributeError: 'function' object has no attribute '_skl_parameter_constraints'

As I understand decorator validate_params add attribute to the function, but in this case we do not use it, so it can't be accessed.

Reply:
I mean we need to have:

```python
@validate_params({
    "X": ["array-like"],
    "assume_centered": ["boolean"],
    "block_size": [Integral],
})
def ledoit_wolf(X, *, assume_centered=False, block_size=1000):
```

and

```python
PARAM_VALIDATION_FUNCTION_LIST = [
    "sklearn.cluster.kmeans_plusplus",
    "sklearn.covariance.ledoit_wolf",
]
```

Doing this on your PR, works for me.

Reply:
Then why not implement the first proposition: add param validation to the functions without rewriting them to use class inside?